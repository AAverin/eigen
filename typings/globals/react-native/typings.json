{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
