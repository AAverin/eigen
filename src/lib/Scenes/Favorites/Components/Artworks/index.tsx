import React, { Component } from "react"
import { ScrollView } from "react-native"
import { createPaginationContainer, graphql, RelayPaginationProp } from "react-relay"

import GenericGrid from "lib/Components/ArtworkGrids/GenericGrid"
import ZeroState from "lib/Components/States/ZeroState"
import { PAGE_SIZE } from "lib/data/constants"
import { isCloseToBottom } from "lib/utils/isCloseToBottom"

interface Props extends RelayProps {
  relay?: RelayPaginationProp
  onDataFetching?: (loading: boolean) => void
}

interface State {
  fetchingMoreData: boolean
}

export class SavedWorks extends Component<Props, State> {
  state = {
    fetchingMoreData: false,
  }

  loadMore = () => {
    if (!this.props.relay.hasMore() || this.props.relay.isLoading()) {
      return
    }

    const updateState = (loading: boolean) => {
      this.setState({ fetchingMoreData: loading })
      if (this.props.onDataFetching) {
        this.props.onDataFetching(loading)
      }
    }

    updateState(true)
    this.props.relay.loadMore(PAGE_SIZE, error => {
      if (error) {
        // FIXME: Handle error
        console.error("Artworks/index.tsx", error.message)
      }
      updateState(false)
    })
  }

  render() {
    const artworks = this.props.me.saved_artworks.artworks_connection.edges.map(edge => edge.node)

    if (artworks.length === 0) {
      return (
        <ZeroState
          title="You havenâ€™t followed any artists yet"
          subtitle="Follow artists to get notified about new works that have been added to Artsy."
        />
      )
    }

    return (
      <ScrollView
        onScroll={isCloseToBottom(this.loadMore)}
        scrollEventThrottle={400}
        style={{ flex: 1 }}
        contentContainerStyle={{ padding: 20 }}
      >
        <GenericGrid artworks={artworks} isLoading={this.state.fetchingMoreData} />
      </ScrollView>
    )
  }
}

export default createPaginationContainer(
  SavedWorks,
  {
    me: graphql.experimental`
      fragment Artworks_me on Me
        @argumentDefinitions(count: { type: "Int", defaultValue: 10 }, cursor: { type: "String", defaultValue: "" }) {
        saved_artworks {
          artworks_connection(private: true, first: $count, after: $cursor)
            @connection(key: "GenericGrid_artworks_connection") {
            pageInfo {
              endCursor
              hasNextPage
            }
            edges {
              node {
                ...GenericGrid_artworks
              }
            }
          }
        }
      }
    `,
  },
  {
    direction: "forward",
    getConnectionFromProps(props) {
      return props.me && props.me.saved_artworks.artworks_connection
    },
    getFragmentVariables(prevVars, totalCount) {
      return {
        ...prevVars,
        count: totalCount,
      }
    },
    getVariables(_props, { count, cursor }, fragmentVariables) {
      return {
        ...fragmentVariables,
        count,
        cursor,
      }
    },
    query: graphql.experimental`
      query ArtworksQuery($count: Int!, $cursor: String) {
        me {
          ...Artworks_me @arguments(count: $count, cursor: $cursor)
        }
      }
    `,
  }
)

interface RelayProps {
  me: {
    saved_artworks: {
      artworks_connection: {
        edges: Array<{
          node: any | null
        }> | null
      } | null
    } | null
  } | null
}
