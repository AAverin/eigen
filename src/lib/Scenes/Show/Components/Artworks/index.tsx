import React from "react"
import { Text } from "react-native"

interface Props {
  artworks: any[]
}

// TODO: implement

export class Artworks extends React.Component<Props> {
  render() {
    return <Text>{JSON.stringify(this.props.artworks)}</Text>
  }
}
