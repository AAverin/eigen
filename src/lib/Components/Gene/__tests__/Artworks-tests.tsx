import "react-native"

import React from "react"
import * as renderer from "react-test-renderer"

import Artworks from "../Artworks"

it("renders properly", () => {
  const props = {
    gene: {
      id: "deep-time",
    },
    medium: "painting",
    queryState: null,
    queryForPage: null,
  }
  const artworks = renderer.create(<Artworks gene={props} />).toJSON()
  expect(artworks).toMatchSnapshot()
})
