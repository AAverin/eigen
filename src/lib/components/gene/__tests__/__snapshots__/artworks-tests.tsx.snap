exports[`test renders properly 1`] = `
<RCTScrollView
  accessibilityLabel="Artworks ScrollView"
  onLayout={[Function]}
  onScroll={[Function]}
  scrollEventThrottle={50}
  scrollsToTop={false}>
  <View>
    <View
      accessibilityLabel="Artworks Content View"
      style={
        Object {
          "flexDirection": "row",
        }
      } />
  </View>
</RCTScrollView>
`;
